data-collection/data-collection/Rich Popup/Attachments/RichPopupStagedAttachment.swift

```diff
@@ -0,0 +1,47 @@
+//// Copyright 2019 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+
+class RichPopupStagedAttachment: NSObject, RichPopupPreviewableAttachment {
+    
+    let attachmentData: Data
+    
+    let attachmentMimeType: String
+    
+    var name: String?
+    
+    init(data: Data, mimeType: String, name: String?) {
+        
+        self.attachmentData = data
+        self.attachmentMimeType = mimeType
+        self.name = name
+        
+        super.init()
+    }
+    
+    /// Optionally, specify a previewItemURL is you would like this staged attachment available to QLPreviewController.
```

Comment:
In addition to @ChrisFrenchPDX 's observation, I would suggest using code formatting for symbols:
```suggestion
    /// Optionally, specify a `previewItemURL` if you would like this staged attachment available to `QLPreviewController`.
```